megamek/src/megamek/client/ratgenerator/RATGenerator.java

```diff
@@ -215,16 +215,22 @@ public AvailabilityRating findModelAvailabilityRecord(int era, String unit, Fact
      * Adds or changes an availability rating entry for a model.
      * 
      * @param era  The year of the record to change
-     * @param unit The name of the model for which to change the record
+     * @param chassisKey The chassis key for the unit which is having its model record updated 
+     * @param unit The model key for the unit which is having its model record updated
      * @param ar   The new <code>AvailabilityRating</code> for the unit in the era. This provides the
      *             faction.
      */
-    public void setModelFactionRating(int era, String unit, AvailabilityRating ar) {
-        if (modelIndex.get(era).get(unit) == null) {
-            modelIndex.get(era).put(unit, new HashMap<String, AvailabilityRating>());
+    public void setModelFactionRating(int era, String chassisKey, String unitKey, AvailabilityRating ar) {
+        // create a blank chassis faction record as well
+        if (chassisIndex.get(era).get(chassisKey) == null) {
+            setChassisFactionRating(era, chassisKey, new AvailabilityRating(chassisKey, era, ar.getFaction() + ":1"));
         }
-        modelIndex.get(era).get(unit).put(ar.getFactionCode(), ar);
-        models.get(unit).getIncludedFactions().add(ar.getFactionCode());
+        
+        if (modelIndex.get(era).get(unitKey) == null) {
+            modelIndex.get(era).put(unitKey, new HashMap<String, AvailabilityRating>());
+        }
+        modelIndex.get(era).get(unitKey).put(ar.getFactionCode(), ar);
+        models.get(unitKey).getIncludedFactions().add(ar.getFactionCode());
```

Comment:
The way it's currently implemented, every model record does have to be accompanied by a chassis record. The chassis saving code simply skips over the individual model records if there is no chassis record. A better approach may be to prevent adding a model record if there is no chassis record.

Reply:
I'll take another look at it when I get the chance and see if there's a better approach.

Reply:
That would be a case of user error, but requiring that there be a chassis record first would guard against it. It might be simpler to validate the data and warn the user in a popup when saving. This was never intended to be a sophisticated tool.